rir/src/compiler/pir/module.h

```diff
@@ -62,7 +64,7 @@ class Module {
 
   private:
     Closure* declare(rir::Function*, const std::vector<SEXP>& a, Env* env);
-    std::unordered_map<rir::Function*, VersionedClosure> functions;
+    std::map<FunctionAndEnv, VersionedClosure> functions;
```

Comment:
yeah, I usually make a mistake when i implement hash<>, so first I wanna see it running with map :) if we care enough to change it, we still can..

Reply:
Oh look, there's a C++ library proposal for `hash_combine`: http://wg21.link/p0814r2

Because the proposals in 2012 and 2014 died...

I'm fine with leaving it like this for now. Do we have any guess as to how big the map would be?